flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/util/HandlerUtils.java

```diff
@@ -108,19 +108,25 @@ public static void sendErrorResponse(
 	 * @param httpRequest originating http request
 	 * @param message which should be sent
 	 * @param statusCode of the message to send
+	 * @param forceClose indicates whether to forcibly close the connection after the response is sent
 	 */
 	public static void sendResponse(
 			@Nonnull ChannelHandlerContext channelHandlerContext,
 			@Nonnull HttpRequest httpRequest,
 			@Nonnull String message,
-			@Nonnull HttpResponseStatus statusCode) {
+			@Nonnull HttpResponseStatus statusCode,
+			boolean forceClose) {
```

Comment:
Would it make sense to not pass forceClose explicitly but to decide this based on `statusCode` and `isServerError` inside of this method?

Reply:
I took this method to be the implementation method having maximum flexibility.  Happy to change it.

Reply:
I would prefer to keep the interface (also internal ones) as lean as possible. Past experience shows that this can otherwise lead to problems down the road (APIs grown too complicated, misuse). If need should arise, then we could change it accordingly.